src/ranch_ssl.erl

```diff
@@ -124,6 +124,15 @@ do_listen(SocketOpts0, Logger) ->
 	SocketOpts2 = ranch:set_option_default(SocketOpts1, nodelay, true),
 	SocketOpts3 = ranch:set_option_default(SocketOpts2, send_timeout, 30000),
 	SocketOpts = ranch:set_option_default(SocketOpts3, send_timeout_close, true),
+        %% In case of a local socket, we remove the socket file first.
+        %% It is possible to have multiple ip tuples in the socket options,
+        %% and the last one will be used (undocumented).
+        _=case lists:keyfind(ip, 1, lists:reverse(SocketOpts)) of
+                {ip, {local, SockFile}} ->
+                        file:delete(SockFile);
+                _ ->
+                        ok
+        end,
```

Comment:
I would do this on `SocketOpts0` instead since there's going to be less options in the list.

Reply:
ok, done, same for `ranch_tcp`